File: examples/nodejs/send-document-message.js
/**
 * Contoh Kode: Mengirim Pesan Document (PDF, DOC, dll)
 * 
 * File ini berisi contoh implementasi untuk mengirim dokumen melalui WhatsApp
 * 
 * Catatan: URL dokumen harus dapat diakses secara publik (public URL)
 */

const { makeRequest, displayResponse } = require('./config');

async function main() {
    console.log('=== WACloud - Send Document Message Example ===\n\n');

    // Device ID yang sudah terhubung (status: connected)
    const deviceId = '550e8400-e29b-41d4-a716-446655440000';

    // Nomor tujuan (format: tanpa leading 0, dengan kode negara)
    const to = '6281234567890';

    // URL dokumen yang dapat diakses publik
    const documentUrl = 'https://example.com/document.pdf';

    // Nama file (akan ditampilkan di WhatsApp)
    const filename = 'document.pdf';

    // Data pesan dokumen
    const messageData = {
        device_id: deviceId,
        to: to,
        message_type: 'document',
        document_url: documentUrl,
        filename: filename,
        caption: 'Ini adalah caption untuk dokumen' // Optional
    };

    console.log('Mengirim pesan dokumen...');
    console.log(`To: ${to}`);
    console.log(`Document URL: ${documentUrl}`);
    console.log(`Filename: ${filename}`);
    console.log(`Caption: ${messageData.caption}\n`);

    const response = await makeRequest('POST', '/messages', messageData);
    displayResponse(response);

    if (response.success && response.data?.message_id) {
        console.log('✓ Pesan dokumen berhasil dikirim!');
        console.log(`Message ID: ${response.data.message_id}`);
        console.log(`Status: ${response.data.status}`);
    } else {
        console.log('✗ Gagal mengirim pesan dokumen.');
        if (response.error) {
            console.log(`Error: ${response.error}`);
        }
        if (response.message) {
            console.log(`Message: ${response.message}`);
        }

        console.log('\nTips:');
        console.log('- Pastikan URL dokumen dapat diakses secara publik');
        console.log('- Format dokumen yang didukung: PDF, DOC, DOCX, XLS, XLSX, PPT, PPTX, dll');
        console.log('- Pastikan device dalam status \'connected\'');
        console.log('- Ukuran file maksimal sesuai limit WhatsApp (biasanya 100MB)');
    }

    console.log('\n=== Selesai ===');
}

main().catch(console.error);

File: examples/python/send_document_message.py
"""
Contoh Kode: Mengirim Pesan Document (PDF, DOC, dll)

File ini berisi contoh implementasi untuk mengirim dokumen melalui WhatsApp

Catatan: URL dokumen harus dapat diakses secara publik (public URL)
"""

from config import make_request, display_response


def main():
    print('=== WACloud - Send Document Message Example ===\n\n')

    # Device ID yang sudah terhubung (status: connected)
    device_id = '550e8400-e29b-41d4-a716-446655440000'

    # Nomor tujuan (format: tanpa leading 0, dengan kode negara)
    to = '6281234567890'

    # URL dokumen yang dapat diakses publik
    document_url = 'https://example.com/document.pdf'

    # Nama file (akan ditampilkan di WhatsApp)
    filename = 'document.pdf'

    # Data pesan dokumen
    message_data = {
        'device_id': device_id,
        'to': to,
        'message_type': 'document',
        'document_url': document_url,
        'filename': filename,
        'caption': 'Ini adalah caption untuk dokumen'  # Optional
    }

    print('Mengirim pesan dokumen...')
    print(f'To: {to}')
    print(f'Document URL: {document_url}')
    print(f'Filename: {filename}')
    print(f"Caption: {message_data['caption']}\n")

    response = make_request('POST', '/messages', message_data)
    display_response(response)

    if response.get('success') and response.get('data', {}).get('message_id'):
        print('✓ Pesan dokumen berhasil dikirim!')
        print(f"Message ID: {response['data']['message_id']}")
        print(f"Status: {response['data']['status']}")
    else:
        print('✗ Gagal mengirim pesan dokumen.')
        if response.get('error'):
            print(f"Error: {response['error']}")
        if response.get('message'):
            print(f"Message: {response['message']}")

        print('\nTips:')
        print('- Pastikan URL dokumen dapat diakses secara publik')
        print('- Format dokumen yang didukung: PDF, DOC, DOCX, XLS, XLSX, PPT, PPTX, dll')
        print('- Pastikan device dalam status \'connected\'')
        print('- Ukuran file maksimal sesuai limit WhatsApp (biasanya 100MB)')

    print('\n=== Selesai ===')


if __name__ == '__main__':
    main()

